src/analysis/seminorm.lean

```diff
@@ -416,6 +484,51 @@ end
 end module
 end normed_linear_ordered_field
 
+section seminorm_sup
+
+noncomputable theory
+
+variables [normed_field 𝕜] [add_comm_group E] [module 𝕜 E] [semi_normed_space ℝ 𝕜]
+variables [module ℝ E]
+variables {ι : Type*} [decidable_eq ι]
+variables (p : ι → seminorm 𝕜 E)
+variables (ι' : finset ι)
+
+@[simp]
+lemma seminorm_sup_singleton (p : ι → seminorm 𝕜 E) (i : ι):
+  ({i} : finset ι).sup p = p i :=
+begin
+  simp,
+end
+
+lemma seminorm_sup_finset_coe_to_fun (p : ι → seminorm 𝕜 E) (ι' : finset ι) :
+  coe_fn (ι'.sup p) = λ x, ↑(ι'.sup (λ i, (p i x).to_nnreal)) :=
```

Comment:
I'd do this in stages, and first prove that
```suggestion
  coe_fn (ι'.sup p) = ι'.sup (λ i, coe_fn (p i)) :=
```

Reply:
this throws an error: failed to synthesize type class instance for `order_bot (E → ℝ)`

Reply:
Oh good point, it doesn't work without the cast to nnreal. So maybe
```
⇑(ι'.sup p) = ι'.sup (λ i x, p i ⟨x, nonneg x⟩)
```

Reply:
still missing the `order_bot (E → ℝ)` instance and it does not like the constructor `⟨...⟩`.

Reply:
this works: `coe_fn (ι'.sup p) = ↑(ι'.sup (λ i x, (p i x).to_nnreal))` but I don't know whether you had that in mind